plugins/outputs/circonus/check_test.go

```diff
@@ -0,0 +1,159 @@
+// Package circonus contains the output plugin used to write metric data to a
+// Circonus broker.
+package circonus
+
+import (
+	"testing"
+
+	apiclient "github.com/circonus-labs/go-apiclient"
+	apiconfig "github.com/circonus-labs/go-apiclient/config"
+)
+
+func stringPtr(s string) *string {
+	return &s
+}
+
+type mockAPIClient struct{}
```

Comment:
Couldn't you just use a [`httptest` server](https://pkg.go.dev/net/http/httptest) here and avoid the "client" interface altogether?